### ISSUE TITLE
Document loading images and masks for deep learning applications

### ISSUE THREAD

**Issue by gkaissis** [loli/medpy#88]
It would be great if some documentation could be provided for how to load a stack of dicoms or a .nii file alongside a segmentation (.nii) and output a cropped n-dimensional array with padding ready for use in deep learning applications.

**Comment by loli** [comment]
I'm not sure if I understand what you want. Something like this?

```python
import numpy as np
from medpy.io import load
from scipy.ndimage import find_objects

def fancyload(fimg, fseg):
  seg, segh = load(fseg)
  seg = seg.astype(np.bool)

  img, imgh = load(fimg)
  img = img.astype(np.float)

  objslices = find_objects(seg)[0]
  # your can add your padding to the returned slice

  return img[objslices], seg[objslices]
```

If not, please elaborate. Cropped to what? Padded by what?

PS: Untested, might contain some typos or bugs

**Comment by gkaissis** [comment]
Say I want to perform a deep learning analysis on a tumor. I open the .nii in ITK-SNAP or some other segmentation software and draw a mask, which I save as a .nii overlay. If I want to enter the image into a deep neural network, I don't want the whole image in there, just the tumor. Therefore I want the program to load the original image, load the mask and throw away everything except the tumor, pad the tumor with zeros and then return an n-dimensional numpy array to put into TensorFlow. I didn't find an explicit mention of such a feature in the documentation but it would be extremely valuable.
Thank you!

**Comment by oskarmaier** [comment]
I am still not 100% sure what you want. Let's assume you have something like this:
![original](https://user-images.githubusercontent.com/34232090/55410663-89e3a980-5564-11e9-91ba-17205123c872.png)

Do you want it to look like this:
![tumor](https://user-images.githubusercontent.com/34232090/55410762-b4cdfd80-5564-11e9-9742-32c952f68277.png)


Like this:
![tumor_cropped](https://user-images.githubusercontent.com/34232090/55410729-a5e74b00-5564-11e9-91a5-365ffc001e1f.png)

Or like this:
![tumor_cropped_not_zero](https://user-images.githubusercontent.com/34232090/55410735-ab449580-5564-11e9-8da6-128646b1e576.png)

All three can be easily achieved by simple numpy masking operations.
